internal/services/healthcare/healthcare_dicom_resource.go

```diff
@@ -373,3 +532,103 @@ func flattenDicomServicePrivateEndpoint(input *[]dicomservices.PrivateEndpointCo
 	}
 	return results
 }
+
+func expandStorageConfiguration(input []interface{}) *dicomservices.StorageConfiguration {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+	storageSettings := input[0].(map[string]interface{})
+
+	var storageResourceId *string
+	if v, ok := storageSettings["storage_account_id"]; ok {
+		storageResourceId = pointer.To(v.(string))
+	}
+
+	var fileSystemName *string
+	if v, ok := storageSettings["file_system_name"]; ok {
+		fileSystemName = pointer.To(v.(string))
+	}
+
+	return &dicomservices.StorageConfiguration{
+		FileSystemName:    fileSystemName,
+		StorageResourceId: storageResourceId,
+	}
+}
+
+func flattenStorageConfiguration(configuration *dicomservices.StorageConfiguration) interface{} {
+	if configuration == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+	if configuration.FileSystemName != nil {
+		result["file_system_name"] = *configuration.FileSystemName
+	}
+
+	if configuration.StorageResourceId != nil {
+		result["storage_account_id"] = *configuration.StorageResourceId
+	}
```

Comment:
We should parse the Storage account ID that we're getting back from the API so that we can catch any casing inconsistencies and potentially prevent them from creeping into user's state
```suggestion
	if v := pointer.From(configuration.StorageResourceId); v != "" {
		id, err := commonids.ParseStorageAccountID(v)
		if err != nil {
			return nil, err
		}
		result["storage_account_id"] = id.ID()
	}
```

Reply:
Thanks, updated the flatten function (used by the resource & data source), the function now also returns  error to the calling function